### PR TITLE
Issues/vat refactoring

### PR DESCRIPTION
**Pull request by AndreyDegtyarev**
Refactored VatValidation.

### CODE REVIEW

File: src/Nox.Reference.Abstractions/VatNumbers/VatValidationStatus.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Nox.Reference.Abstractions;
+
+public enum VatValidationStatus
+{
+    NotVerified = 1,
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Abstractions/VatNumbers/VatValidationStatus.cs:5]: Should we add "Unknown = 0" as default?

File: src/Nox.Reference.Data.World/Models/VatNumbers/VatNumberValidationResult.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            result.AddError(validationError);
+        }
+
+        result.Status = VatValidationStatus.NotVerified;
+        return result;
+    }
+
+    public static VatNumberValidationResult CreateWithValidaton(string vatNumber, string country)
+    {
+        return CreateWithValidaton(VatValidationStatus.Valid, vatNumber, country);
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Data.World/Models/VatNumbers/VatNumberValidationResult.cs:28]: Why status is always valid?

Comment (ILoveMyTortoise) [reply src/Nox.Reference.Data.World/Models/VatNumbers/VatNumberValidationResult.cs:28]: Change method name or code

File: src/Nox.Reference.Data.World/Models/VatNumbers/VatNumberValidationResult.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        {
+            OriginalVatNumber = vatNumber,
+            FormattedVatNumber = NormalizeVatNumber(vatNumber, country),
+            Status = status,
+            Country = country
+        };
+        return result;
+    }
+
+    public VatValidationStatus Status { get; private set; } = VatValidationStatus.NotVerified;
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Data.World/Models/VatNumbers/VatNumberValidationResult.cs:46]: why not on top?

File: src/Nox.Reference.Abstractions/VatNumbers/VatValidationStatus.cs (hunk 1 above)
Comment (AndreyDegtyarev) [reply src/Nox.Reference.Abstractions/VatNumbers/VatValidationStatus.cs:5]: I guess Unknown is the same as NotVerified ?

File: src/Nox.Reference.Data.World/Models/VatNumbers/VatNumberValidationResult.cs (hunk 2 above)
Comment (AndreyDegtyarev) [reply src/Nox.Reference.Data.World/Models/VatNumbers/VatNumberValidationResult.cs:28]: CreateWithoutValidation - validation is not carried out, result status is NotVerified (or might be Unknown)
CreateWithValidaton - If validation carries out it marks result as valid by default until error happens

File: src/Nox.Reference.Abstractions/VatNumbers/VatValidationStatus.cs (hunk 1 above)
Comment (andresharpe) [reply src/Nox.Reference.Abstractions/VatNumbers/VatValidationStatus.cs:5]: I think UnVerified is clearer and less ambiguous. 

"This VAT info is unverified"

Is better than

"This VAT info is unknown"

Comment (AndreyDegtyarev) [reply src/Nox.Reference.Abstractions/VatNumbers/VatValidationStatus.cs:5]: So, just leave as is. NotVerified )

Comment (AndreyDegtyarev) [reply src/Nox.Reference.Abstractions/VatNumbers/VatValidationStatus.cs:5]: Set Unverified

File: data/output/raw-json/VatNumberDefinitions/Nox.Reference.VatNumberDefinitions.json [hunk 4]
```diff
@@ -0,0 +1,836 @@
...
+          "modulus": null,
+          "weights": null
+        }
+      }
+    ],
+    "verificationApi": "EuropeVies"
+  },
+  {
+    "country": "RU",
+    "localName": "\u0418\u0434\u0435\u043D\u0442\u0438\u0444\u0438\u043A\u0430\u0446\u0438\u043E\u043D\u043D\u044B\u0439 \u043D\u043E\u043C\u0435\u0440 \u043D\u0430\u043B\u043E\u0433\u043E\u043F\u043B\u0430\u0442\u0435\u043B\u044C\u0449\u0438\u043A\u0430",
```
Comment (ILoveMyTortoise) [inline data/output/raw-json/VatNumberDefinitions/Nox.Reference.VatNumberDefinitions.json:727]: hm, why is it broken?

File: src/Nox.Reference.Data.World/Configurations/LanguageConfiguration.cs [hunk 5]
```diff
@@ -8,7 +8,7 @@ internal class LanguageConfiguration : NoxReferenceEntityConfigurationBase<Langu
     protected override void DoConfigure(EntityTypeBuilder<Language> builder)
     {
         builder
-            .Property(x => x.Name)
-            .IsRequired();
+            .HasMany(x => x.NameTranslations)
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Data.World/Configurations/LanguageConfiguration.cs:11]: let's have a small chat on it

**Review by ILoveMyTortoise — APPROVED** [review]

File: data/output/raw-json/VatNumberDefinitions/Nox.Reference.VatNumberDefinitions.json (hunk 4 above)
Comment (AndreyDegtyarev) [reply data/output/raw-json/VatNumberDefinitions/Nox.Reference.VatNumberDefinitions.json]: Set Encoding Unicode

File: src/Nox.Reference.Data.World/Configurations/LanguageConfiguration.cs (hunk 5 above)
Comment (AndreyDegtyarev) [reply src/Nox.Reference.Data.World/Configurations/LanguageConfiguration.cs:11]: TODO:

**Pull request merged**